openmls/src/group/mls_group/test_mls_group.rs

```diff
@@ -1195,3 +1208,86 @@ fn builder_pattern(ciphersuite: Ciphersuite, provider: &impl OpenMlsProvider) {
         .expect_err("successfully built group with invalid leaf extensions");
     assert_eq!(builder_err, InvalidExtensionError::IllegalInLeafNodes);
 }
+
+// Test that unknown group context and leaf node extensions can be used in groups
+#[apply(ciphersuites_and_providers)]
+fn unknown_extensions(ciphersuite: Ciphersuite, provider: &impl OpenMlsProvider) {
+    let (alice_credential_with_key, _alice_kpb, alice_signer, _alice_pk) =
+        setup_client("Alice", ciphersuite, provider);
+
+    let unknown_gc_extension = Extension::Unknown(0xff00, UnknownExtension(vec![0, 1, 2, 3]));
```

Comment:
You only use private IDs here. Which is good.
I wonder if we should enforce that, or allow any ID for unknown extensions.

Reply:
Turns out this is a bit trickier than I expected and I'd rather not do it in this PR. I've added an issue, though: #1480.